ros2action/ros2action/verb/send_goal.py

```diff
@@ -148,28 +176,6 @@ def send_goal(action_name, action_type, goal_values, feedback_callback):
         print('Result:\n    {}'.format(message_to_yaml(result.result)))
         print('Goal finished with status: {}'.format(_goal_status_to_string(result.status)))
     finally:
-        # Cancel the goal if it's still active
-        if (goal_handle is not None and
```

Comment:
The effect of adding _sigint_cancel_handler is the same as adding "`rclpy.ok() and `" here.   
Is it possible to modify this directly ?

Reply:
No i do not think so, if there is no signal handler, `rclpy::shutdown_context` will be called in another thread via signal, and that is gonna be the racy condition.

Reply:
Yes. This avoids the racy condition.   

I just thought of another case. 

user press ctrl +c while spin_until_future_complete() is executed. 

https://github.com/ros2/ros2cli/blob/35d6abd50a55181099aab82134fc507d8fd162bd/ros2action/ros2action/verb/send_goal.py#L119-L122

At this time, rclpy::shutdown_context() will be called. (That is, the default context is shutdown)  

And user press ctrl +c again after running signal.signal(signal.SIGINT, _sigint_cancel_handler). `_sigint_cancel_handler` will be called. At this time, the issue also occurs (rclpy.spin_until_future_complete(node, cancel_future at line 135 directly return since context isn't ok).   
Whether do we need to add "` rclpy.ok() and `"  at line 130 ?

Reply:
we can add that for just a precaution, but i do not think that is necessary.

this signal handler will be called exclusively on 1st SIGINT, if user keep sending the SIGINT, that will be queued in the kernel until the 1st handler completes the function. at this moment, `goal_handle` status will be `STATUS_CANCELED` so SIGINT signal handler does nothing and just gets out of the loop.